pkg/virtctl/portforward/portforward.go

```diff
@@ -0,0 +1,197 @@
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	forwardToStdoutFlag = "forwardToStdout"
+	addressFlag         = "address"
+)
+
+var (
+	forwardToStdout bool
+	address         string = "127.0.0.1"
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "port-forward (VM|VMI)",
+		Short:   "Forward local ports to a virtual machine instance.",
+		Example: usage(),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if n := len(args); n < 2 {
+				fmt.Printf("fatal: Number of input parameters is incorrect, portforward requires at least 2 arg(s), received %d\n\n", n)
+				cmd.Help()
+				return errors.New("argument validation failed")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := PortForward{clientConfig: clientConfig}
+			return c.Run(cmd, args)
+		},
+	}
+	cmd.Flags().BoolVar(&forwardToStdout, forwardToStdoutFlag, forwardToStdout,
+		fmt.Sprintf("--%s=true: Set this to true to forward the tunnel to stdout/stdin; Only works with a single port", forwardToStdoutFlag))
+	cmd.Flags().StringVar(&address, addressFlag, address,
+		fmt.Sprintf("--%s=: Set this to the addres the local ports should be opened on", addressFlag))
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+type PortForward struct {
+	address      *net.IPAddr
+	clientConfig clientcmd.ClientConfig
+	resource     portforwardableResource
+}
+
+func (o *PortForward) Run(cmd *cobra.Command, args []string) error {
+	kind, namespace, name, ports, err := o.prepareCommand(args)
+	if err != nil {
+		return err
+	}
+
+	if err := o.setResource(kind, namespace); err != nil {
+		return err
+	}
+
+	if forwardToStdout {
+		if len(ports) != 1 {
+			return errors.New("only one port supported when forwarding to stdout")
+		}
+		return o.startStdoutStream(namespace, name, ports[0])
+	}
+
+	o.address, err = net.ResolveIPAddr("", address)
+	if err != nil {
+		return err
+	}
+
+	if err := o.startPortForwards(namespace, name, ports); err != nil {
+		return err
+	}
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	<-c
+
+	return nil
+}
+
+func (o *PortForward) prepareCommand(args []string) (kind string, namespace string, name string, ports []forwardedPort, err error) {
+	kind, namespace, name, err = templates.ParseTarget(args[0])
+	if err != nil {
+		return
+	}
+
+	ports, err = parsePorts(args[1:])
+	if err != nil {
+		return
+	}
+
+	if len(namespace) < 1 {
+		namespace, _, err = o.clientConfig.Namespace()
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+func (o *PortForward) setResource(kind, namespace string) error {
+	client, err := kubecli.GetKubevirtClientFromClientConfig(o.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	if kind == "vmi" {
+		o.resource = client.VirtualMachineInstance(namespace)
+	} else if kind == "vm" {
+		o.resource = client.VirtualMachine(namespace)
+	}
+
+	return nil
```

Comment:
I think the validation for unknown types may be missing here.

Reply:
yeah, I relied on the previous validation here. Adding it just to be sure.